src/util/stop.c

```diff
@@ -49,17 +75,21 @@ int nlopt_stop_f(const nlopt_stopping * s, double f, double oldf)
 int nlopt_stop_x(const nlopt_stopping * s, const double *x, const double *oldx)
 {
     unsigned i;
+    if (diff_norm(s->n, x, oldx, s->x_weights) <= s->xtol_rel * diff_norm(s->n, x, oldx, s->x_weights))
```

Comment:
You are comparing `diff_norm < xtol_rel * diff_norm`, which seems wrong — it is equivalent to checking `1 <= xtol_rel`, no?

Should probably be:
```c
if (diff_norm(s->n, x, oldx, s->x_weights) <= s->xtol_rel * vector_norm(s->n, x, s->x_weights))
```
?